lib/mix/lib/mix/tasks/escript.build.ex

```diff
@@ -319,6 +319,7 @@ defmodule Mix.Tasks.Escript.Build do
 
         defp load_config(config) do
           each_fun = fn {app, kw} ->
+            :application.load(app)
```

Comment:
So theoretically you don't need to load the application if you are using the `persistent: true`, as `persistent: true` gives it higher precedence:

```
iex(1)> Application.get_all_env :ex_unit
[]
iex(2)> Application.put_env(:ex_unit, :stacktrace_depth, 30, persistent: true)
:ok
iex(3)> Application.get_all_env :ex_unit
[stacktrace_depth: 30]
iex(4)> Application.load :ex_unit
:ok
iex(5)> Application.get_all_env :ex_unit
[
  exclude: [],
  trace: false,
  slowest: 0,
  after_suite: [],
  assert_receive_timeout: 100,
  formatters: [ExUnit.CLIFormatter],
  autorun: true,
  max_failures: :infinity,
  capture_log: false,
  refute_receive_timeout: 100,
  timeout: 60000,
  colors: [],
  stacktrace_depth: 30,
  include: [],
  module_load_timeout: 60000
]
```

In the example above, the default value for the stacktrace depth is 20.